src/applause-button.js

```diff
@@ -109,42 +117,26 @@ class ApplauseButton extends HTMLElement {
         );
         // send the updated clap count - checking the response to see if the server-held
         // clap count has actually incremented
-        updateClaps(this.api, increment, this.url).then((updatedClapCount) => {
-          if (updatedClapCount === this._cachedClapCount) {
-            // if the clap number was not incremented, disable further updates
-            this.classList.add("clap-limit-exceeded");
-            // and reset the counter
-            this._countElement.innerHTML = formatClaps(updatedClapCount);
-          }
-          this._cachedClapCount = updatedClapCount;
-
-          this._totalClaps += increment;
-          this._bufferedClaps = 0;
-        });
+        updateClaps(this.api, increment, this.url)
+          .then((updatedClapCount) => {
+            if (updatedClapCount === this._cachedClapCount) {
+              // if the clap number was not incremented, disable further updates
+              this._setClapLimitReached();
+              // and reset the counter
+              this._countElement.innerHTML = formatClaps(this._cachedClapCount);
+            } else {
+              this._cachedClapCount = updatedClapCount;
+              this._totalClaps += increment;
+            }
+            this._bufferedClaps = 0;
+          })
+          .catch(console.error);
       }
     }, 2000);
 
-    this.addEventListener("mousedown", (event) => {
```

Comment:
Do we know what the motivation was for having `mousedown` originally? We're changing the behaviour slightly (very slightly) by setting the event to `click`

Reply:
makes sense 👍